code/ui/blocks/src/controls/Files.tsx

```diff
@@ -48,6 +48,17 @@ export const FilesControl: FC<FilesControlProps> = ({
     revokeOldUrls(value);
   }
 
+  // Added useEffect hook to reset the file value when value is null
+  useEffect(() => {
+    if (value == null) {
+      const input = document.getElementById(getControlId(name));
```

Comment:
Could you use a `ref` to the element instead of getting it from `document`?

In theory there could be an ID collision, which would make this misbehave - don't know if there could be in practice.
But using a `ref` would be more React-y anyway, so let's do that if it's as simple.